### PR TITLE
Allow custom parser in Request [was: Fixes #740]

### PR DESCRIPTION
**Pull request by alex94puchades**
Added parse method to Request class instead of Response

### CODE REVIEW

**Comment by defunctzombie** [comment]
What is the issue? Please make a better commit message. Was the previous commit not tested? Please add tests for this now, clearly the previous commit was bad and I should have been more thorough in reviewing it.

File: lib/client.js [hunk 1]
```diff
@@ -980,7 +980,8 @@ Request.prototype.end = function(fn){
   if ('GET' != this.method && 'HEAD' != this.method && 'string' != typeof data && !isHost(data)) {
     // serialize stuff
     var contentType = this.getHeader('Content-Type');
-    var serialize = request.serialize[contentType ? contentType.split(';')[0] : ''];
+    var contentParser = request.serialize[contentType ? contentType.split(';')[0] : ''];
```
Comment (defunctzombie) [inline lib/client.js]: `this._parser || ` can just prefix this to avoid the lookup if we have a parser

**Comment by alex94puchades** [comment]
@defunctzombie Sorry, really, it was my bad. This is what happens when "fast" gets in the place of "good". I am really sorry. The issue with the last commit was that it added the ability to override a parser for the Response, when what I wanted to was to add that ability to Request. It WAS tested, it was tested not to break anything that wasn't broken, and I was adding unit tests sometime in the future. Everything I said still applies: no backward-incompatible changes, no broken tests, it was just a typo.

**Comment by defunctzombie** [comment]
Can you add a client test for these changes so it doesn't break in the
future and we know it works plz :)

File: lib/client.js (hunk 1 above)
Comment (alex94puchades) [inline lib/client.js]: Is that a real issue? Sorry, I was just trying to keep within the 80 character line width

Comment (defunctzombie) [inline lib/client.js]: Nope, just a suggestion.

**Comment by alex94puchades** [comment]
Wow, seems like this is gonna take a little more time than I thought : /

**Comment by alex94puchades** [comment]
I fear this is gonna take me some time : /

**Comment by alex94puchades** [comment]
@defunctzombie pffew, all tests pass now :)

**Comment by alex94puchades** [comment]
@defunctzombie Ready to be merged at last! ^^

**Comment by defunctzombie** [comment]
please remove the dummy commit and squash remaining commits into a single one with a commit message describing the change

**Comment by alex94puchades** [comment]
Done

**Comment by defunctzombie** [comment]
@alex94puchades what's with the "fix detection of root object" commit?

**Comment by alex94puchades** [comment]
Sorry, I think it got included while squashing commits, I hope everything is fine now

**Comment by defunctzombie** [comment]
:+1:

**Pull request merged**